### ISSUE TITLE
Option 'throw' is broken or badly documented

### ISSUE THREAD

**Issue by marklagendijk** [grafana/k6-docs#251]
## Expected behaviour
Enabling [throw](https://docs.k6.io/docs/options) exits the specific test iteration as soon as an request error occurs: connection issues, status 4xx, status 5xxx.

I hoped that with this option I wouldn't have to write `check`s for every request.

## Actual behaviour
Status errors don't exit the iteration, or show anything in the log.

## Documented behaviour
A boolean, true or false, specifying whether to throw errors on failed HTTP requests or not.

**Comment by simskij** [comment]
@na-- should this really be here? I get the impression that you wanted to have this as some kind of input to the design of the new HTTP API?

**Comment by na--** [comment]
A bit of both. We should better document how the current `throw` option works, because it's unfortunately not going to disappear anytime soon and it has some behaviors that, while mostly logically consistent, are still somewhat surprising. For example, the fact that it won't throw an error for a 4xx or 5xx HTTP response, but will for a network failure. 

Or that, until recently, if you had mistyped your URL, k6 would have thrown an error, even if `throw` was `false`. This was a bug and has now fixed in https://github.com/k6io/k6/pull/2045 and will be released in k6 v0.33.0 at the end of the month, but it probably still deserves a mention, since the options page wasn't part of the versioning done in https://github.com/k6io/docs/pull/283.

And yes, we'll probably rethink if we'll even have a `throw` option in that new eventual HTTP API. But that has been on the back of the backlog for at least 2 years now, and it may not land any time soon, even in an xk6 extension... :disappointed: And even when it lands, we probably will still have to support the `throw` option for a while longer in the old API, just for backwards compatibility... 

In summary: this issue should be here, and it probably deserves at least a normal priority, since it's a common footgun.

**Comment by simskij** [comment]
Appreciated, thanks!

**Comment by ppcano** [comment]
It has been three years since the creation of this issue. 

Now, [k6chaijs](https://k6.io/docs/javascript-api/jslib/k6chaijs/) tries to solve this problem by throwing an assertion on failures. 

I am closing this issue. I suggest following the discussion on the k6 repo or https://github.com/grafana/k6-jslib-k6chaijs/
